dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/UploadWorker.kt

```diff
@@ -33,38 +32,34 @@ internal class UploadWorker(
 
     @WorkerThread
     override fun doWork(): Result {
-        if (!Datadog.isInitialized()) {
+        // the idea behind upload is the following:
+        // 1. we shuffle features list to randomize initial upload task sequence. It is done to
+        // avoid the possible bottleneck when some feature has big batches which are uploaded
+        // slowly, so that next time other features don't wait and have a chance to go before.
+        // 2. we introduce FIFO queue also to avoid the bottleneck: if some feature batch cannot
+        // be uploaded we put retry task to the end of queue, so that batches of other features
+        // have a chance to go.
+        val sdkCore = Datadog.getInstance() as? InternalSdkCore
```

Comment:
When we create `uploadWorkRequest` we can add the name of the instance it is created for and then read it here and get the instance needed.